sbncode/Calibration/TrackCaloSkimmer_module.cc

```diff
@@ -545,14 +545,14 @@ geo::Point_t WireToTrajectoryPosition(const geo::Point_t &loc, const geo::TPCID
 sbn::TrueParticle TrueParticleInfo(const simb::MCParticle &particle,
     const std::vector<geo::BoxBoundedGeo> &active_volumes,
     const std::vector<std::vector<geo::BoxBoundedGeo>> &tpc_volumes,
-    const std::map<int, std::vector<std::pair<geo::WireID, const sim::IDE *>>> &id_to_ide_map,
+    const std::map<int, std::vector<caf::ParticleIDE>> &id_to_ide_map,
     const std::map<int, std::vector<art::Ptr<recob::Hit>>> &id_to_truehit_map, 
     const detinfo::DetectorPropertiesData &dprop,
     const geo::GeometryCore *geo,
     const geo::WireReadoutGeom *wireReadout) {
 
-  std::vector<std::pair<geo::WireID, const sim::IDE *>> empty;
-  const std::vector<std::pair<geo::WireID, const sim::IDE *>> &particle_ides = id_to_ide_map.count(particle.TrackId()) ? id_to_ide_map.at(particle.TrackId()) : empty;
+  std::vector<caf::ParticleIDE> empty;
+  const std::vector<caf::ParticleIDE> &particle_ides = id_to_ide_map.count(particle.TrackId()) ? id_to_ide_map.at(particle.TrackId()) : empty;
```

Comment:
The current implementation performs the same query twice, and that would be better avoided. A way to avoid it, wrapped in a lambda function:
```suggestion
  auto trackReadoutIDEs = [&id_to_ide_map](int trackID) -> std::vector<caf::ParticleIDE> const&
    {
      static std::vector<caf::ParticleIDE> const empty;
      auto const itIDEs = id_to_ide_map.find(trackID);
      return (itIDEs == id_to_ide_map.end())? empty: itIDEs->second;
     };
   
  std::vector<caf::ParticleIDE> const& particle_ides = trackReadoutIDEs(particle.TrackId());
```
This is a _suggestion_.

Reply:
I think it's ok as is -- this would be a small optimization at the cost of readability.